src/actions/hasRole.ts

```diff
@@ -0,0 +1,29 @@
+import { Address, Chain, PublicClient, ReadContractReturnType, Transport } from 'viem';
+import { upgradeExecutor } from '../contracts';
+import { ActionParameters } from '../types/Actions';
+import { UpgradeExecutorRole } from '../upgradeExecutorEncodeFunctionData';
+
+type Args = {
+  role: UpgradeExecutorRole;
+  address: Address;
+};
+type UpgradeExecutorABI = typeof upgradeExecutor.abi;
+export type HasRoleParameters<Curried extends boolean = false> = ActionParameters<
+  Args,
+  'upgradeExecutor',
+  Curried
+>;
+
+export type HasRoleReturnType = ReadContractReturnType<UpgradeExecutorABI, 'hasRole'>;
+
+export async function hasRole<TChain extends Chain | undefined>(
+  client: PublicClient<Transport, TChain>,
+  args: HasRoleParameters,
```

Comment:
This will be changed once we merge https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/140

Then it will use the same mechanism as rollupAdmin getters where we either accept `upgradeExecutor` or `sequencerInbox` and fetch `upgradeExecutor` contract if needed